test/Feature/InsertSpec.hs

```diff
@@ -92,55 +114,99 @@ spec = afterAll_ resetDb $ around withApp $ do
     context "jsonb" . after_ (clearTable "json") $ do
       it "serializes nested object" $ do
         let inserted = [json| { "data": { "foo":"bar" } } |]
-        p <- request methodPost "json" [("Prefer", "return=representation")] inserted
-        liftIO $ do
-          simpleBody p `shouldBe` inserted
-          simpleHeaders p `shouldSatisfy` matchHeader hLocation "/json\\?data=eq\\.%7B%22foo%22%3A%22bar%22%7D"
-          simpleStatus p `shouldBe` created201
+        request methodPost "/json"
+                     [("Prefer", "return=representation")]
+                     inserted
+          `shouldRespondWith` ResponseMatcher {
+            matchBody    = Just inserted
+          , matchStatus  = 201
+          , matchHeaders = ["Location" <:> [str|/json?data=eq.{"foo":"bar"}|]]
+          }
+
+        -- TODO! the test above seems right, why was the one below working before and not now
```

Comment:
I guess what matters is that the url in the location header is valid for later GET requests. Actually it seems like neither the raw value nor the encoded value work.

Both of these urls `http://localhost:3000/json?data=eq.%7B%22foo%22%3A%22bar%22%7D`, `http://localhost:3000/json?data=eq.{"foo":"bar"}` give me the result below.

```json
{
  "hint": "No operator matches the given name and argument type(s). You might need to add explicit type casts.",
  "details": null,
  "code": "42883",
  "message": "operator does not exist: json = unknown"
}
```